common/src/main/org/jetbrains/lincheck/trace/TraceUtils.kt

```diff
@@ -72,4 +72,32 @@ fun String.removeLeadingDollar(): String = removePrefix("$")
 
 fun String.removeVolatileDollar(): String = removeSuffix("\$volatile")
 
-fun String.removeVolatileDollarFU(): String = removeSuffix("\$volatile\$FU")
\ No newline at end of file
+fun String.removeVolatileDollarFU(): String = removeSuffix("\$volatile\$FU")
+
+/**
+ * Used for compressing default pairs and access pairs.
+ */
+fun haveSameOwner(currentClassName: String, nextClassName: String): Boolean =
+    currentClassName == nextClassName
```

Comment:
After refactoring, having `haveSameOwner` as separate method kinda makes no sense, because it is just equality comparison.